src/Symfony/Bridge/Monolog/Handler/FingersCrossed/HttpCodeActivationStrategy.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Handler\FingersCrossed;
+
+use Monolog\Handler\FingersCrossed\ErrorLevelActivationStrategy;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Activation strategy that ignores certain HTTP codes.
+ *
+ * @author Shaun Simmons <shaun@envysphere.com>
+ */
+class HttpCodeActivationStrategy extends ErrorLevelActivationStrategy
```

Comment:
It should not extend `ErrorLevelActivationStrategy` but rather implement `ActivationStrategyInterface`, since this strategy doesn't deal with the error levels.

Reply:
Plus, this class could be `final` IMO.

Reply:
@pimolo The `FingersCrossedHandler` leaves it up to the activation strategy object to handle the action level. It defaults to the `ErrorLevelActivationStrategy` only when it isn't explicitly given an object implementing `ActivationStrategyInterface`. Because the HttpCodeActivationStrategy is given to the FingersCrossedHandler when the `excluded_http_codes` configuration exists, it becomes responsible for handling the action level, just like the existing `NotFoundActivationStrategy` does. So, that's why it's extending the `ErrorLevelActivationStrategy` class.